oracles/engine.go

```diff
@@ -117,13 +117,15 @@ func (e *Engine) BroadcastData(ctx context.Context, data OracleData) error {
 // OracleData matches the spec.
 // It returns a SubscriptionID that is used to Unsubscribe.
 // If cb is nil, the method panics.
-func (e *Engine) Subscribe(ctx context.Context, spec OracleSpec, cb OnMatchedOracleData) SubscriptionID {
+func (e *Engine) Subscribe(ctx context.Context, spec OracleSpec, cb OnMatchedOracleData) (SubscriptionID, Unsubscriber) {
 	if cb == nil {
 		panic(fmt.Sprintf("a callback is required for spec %v", spec))
 	}
 	updatedSubscription := e.subscriptions.addSubscriber(spec, cb, e.timeService.GetTimeNow())
 	e.sendNewOracleSpecSubscription(ctx, updatedSubscription)
-	return updatedSubscription.subscriptionID
+	return updatedSubscription.subscriptionID, func() {
```

Comment:
The context contains the block-height which is sent along with all events. So I think instead of having it part of the closure, you'll need to pass it in when you call in otherwise the oracle-event will have an old block-height associated with it.

Reply:
Yes, I made the last changes to use the same context: https://github.com/vegaprotocol/vega/pull/5557/commits/5a6e9af45a91d27271e05409cafc9e50c643e297#diff-9030c1f6e5bde7618ca69f3b4f5ffc9c9b3cc31cac3ef5a9fb599074bb7808bfR126-R128
Thanks!